pkg/image/nbdkit.go

```diff
@@ -228,6 +236,30 @@ func getVddkPluginPath() NbdkitPlugin {
 	return NbdkitVddkPlugin
 }
 
+// Extra VDDK configuration options are stored in a ConfigMap mounted to the
+// importer pod. Just look for the first file in the mounted directory, and
+// pass that through nbdkit via the "config=" option.
+func getVddkConfig() (string, error) {
+	withHidden, err := os.ReadDir(common.VddkArgsDir)
+	if err != nil {
+		if os.IsNotExist(err) { // No extra arguments ConfigMap specified, so mount directory does not exist
+			return "", nil
+		}
+		return "", err
+	}
+	files := []fs.DirEntry{}
+	for _, file := range withHidden { // Ignore hidden files
```

Comment:
hmm, how come mounting a config map generates hidden files?

Reply:
The volume mount sets up the directory like this:
```
drwxr-sr-x. 2 root  107 4096 Dec 18 15:55 ..2024_12_18_15_55_18.1193095340
lrwxrwxrwx. 1 root  107   32 Dec 18 15:55 ..data -> ..2024_12_18_15_55_18.1193095340
lrwxrwxrwx. 1 root  107   18 Dec 18 15:55 vddk-key -> ..data/vddk-key
```
I don't really know why, I didn't look into the implementation details. The documented way to use it is to open the file with the name of the key from the ConfigMap, so I was just skipping the entries with leading dots.

I guess it would be nicer to pass down the name of the key somehow, but what I was really trying to do was get the ConfigMap contents passed down in a way that CDI doesn't have to parse any of it.

Reply:
You could do something like this https://github.com/kubevirt/containerized-data-importer/blob/625a9e944e3b75e514ca67e394c3e9931bf9634a/pkg/operator/resources/namespaced/controller.go#L342-L355

Reply:
This would be nice, but I was trying to not force a specific key name. With KeyToPath I would need to open the ConfigMap and look for the first key. Maybe it makes sense to just require a specific key name and avoid this issue entirely?

Reply:
Yeah sounds cleaner.. whatever you prefer